### DISCUSSION TITLE
Request parser with fallback

### DISCUSSION THREAD

**Discussion by akaNightmare** [tokio-rs/axum#2839] [Q&A]
### Summary

Hello, I'm wondering if there any way to parse request with a fallback, example:
```rust
pub MyStruct {
  pub field1: bool,
  pub field2: u32,
  pub field3: String,
}
```
so for field1 need to extract value from query params `field1=true` or from headers['x-field1'] and so on.

Thanks in advance.

### axum version

0.7.5

**Comment by mladedav** [comment]
I don't think there's currently a way to parse something from a query parameter or from header. You could write your own type implementing `FromRequestParts` which would probably just wrap the two inner extractors. Then you could have three extractors, one for each of the fields.

Or you can write one extractor specifically for this usecase that tries query and then headers for each of your fields.
